tsc-lsif/src/lsif.ts

```diff
@@ -362,6 +363,7 @@ abstract class SymbolItem {
 	public resultSet: ResultSet;
 	public definitionResult: DefinitionResult;
 	public referenceResult:  ReferenceResult;
+	public implementationResult: ImplementationResult;

```

Comment:
I am not sure if we should add this to all SymbolItems. From my understanding only classes, interface and method have this. Other symbols like local parameters, function, ... do not have an implementation result so we shouldn't send memory for it nor should we emit it.

Reply:
Actually, local variables can have implementation results. There are some interesting examples in [this PR](https://github.com/Microsoft/TypeScript/pull/10482). The following snippet:

`let foo = (a, b) => a + b;
let c = foo(1, 2);`

If you go to implementation on foo ref (line 2) it will point you back to foo def (line 1).

Also, in the following example:

`interface Base {
    hello(): void;
}

let foo: Base = { hello: () => { console.log("Hello!"); } };`

The `foo` variable is considered an implementation of `Base`.

There are a lot of unusual cases we need to cover.

Reply:
Very good catch